sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy

```diff
@@ -15,41 +19,58 @@ import com.vaadin.ui.themes.ValoTheme
  *
  * @since 1.0.0
  *
-*/
-class ProjectOverviewView extends VerticalLayout{
+ */
+class ProjectOverviewView extends VerticalLayout {
 
     private Label titleLabel
     private ProjectOverviewViewModel viewModel
     private Grid<ProjectSummary> projectGrid
+    private final Map<ProjectSummary, DownloadProjectComponent> downloadComponents
+    private final Consumer<String> projectDownload
 
     final static int MAX_CODE_COLUMN_WIDTH = 400
     final static int MAX_STATUS_COLUMN_WIDTH = 200
 
-    ProjectOverviewView(ProjectOverviewViewModel viewModel){
+    ProjectOverviewView(ProjectOverviewViewModel viewModel) {
         this.viewModel = viewModel
-
+        this.downloadComponents = new HashMap<>()
+        this.projectDownload = {
+            //TODO call download controller here
+            projectCode -> println("Starting download of $projectCode")
+        }
         initLayout()
         fillGrid()
     }
 
-    private void initLayout(){
+    private void initLayout() {
         titleLabel = new Label("Project Overview")
         titleLabel.addStyleName(ValoTheme.LABEL_LARGE)
         projectGrid = new Grid<>()
 
+        //FIXME remove test button
+        Button button = new Button("Generate")
+        button.addClickListener({
+            viewModel.updateCount("QSTTS", new Random().nextInt(100))
+        })
+        this.addComponent(button)
+        //end of fixme
         this.addComponents(titleLabel, projectGrid)
     }
 
-    private void fillGrid(){
-        projectGrid.addColumn({ it.code})
-                .setCaption("Project Code").setId("ProjectCode").setMaximumWidth(MAX_CODE_COLUMN_WIDTH)
+    private void fillGrid() {
+        projectGrid.addColumn({ it.code })
+                .setCaption("Project Code").setId("ProjectCode").setMaximumWidth(
+                MAX_CODE_COLUMN_WIDTH)
```

Comment:
I think the width needs a little bit of adjustment
<img width="244" alt="Bildschirmfoto 2021-08-23 um 14 59 01" src="https://user-images.githubusercontent.com/29627977/130451353-fca37281-b332-4f46-8591-55f4daf8478c.png">
Even on a bigger screen the overflow occurs(although the table is not spaced correctly so maybe that has to be fixxed first):
![Bildschirmfoto 2021-08-23 um 15 00 50](https://user-images.githubusercontent.com/29627977/130451655-4fb87bb7-3e0d-4ca8-ba86-0d7f3b3ac198.png)